dateparser/date.py

```diff
@@ -182,21 +182,22 @@ def __init__(self, locale, date_string, date_formats, settings=None):
         self.date_formats = date_formats
         self._translated_date = None
         self._translated_date_with_formatting = None
+        self._parsers = {
+            'timestamp': self._try_timestamp,
+            'relative-time': self._try_freshness_parser,
+            'custom-formats': self._try_given_formats,
+            'absolute-time': self._try_parser,
+            'base-formats': self._try_hardcoded_formats,
+        }
 
     @classmethod
     def parse(cls, locale, date_string, date_formats=None, settings=None):
         instance = cls(locale, date_string, date_formats, settings)
         return instance._parse()
 
     def _parse(self):
-        for parser in (
-            self._try_timestamp,
-            self._try_freshness_parser,
-            self._try_given_formats,
-            self._try_parser,
-            self._try_hardcoded_formats,
-        ):
-            date_obj = parser()
+        for parser in self._settings.PARSERS:
+            date_obj = self._parsers[parser]()
```

Comment:
If we add a non existent parser to the settings this will raise `KeyError`. 

I think that adding a warn like this:
```python
    def _parse(self):
        for parser in self._settings.PARSERS:
            parser_fnx = self._parsers.get(parser)
            if not parser_fnx:
                warn('"{}" is not a valid parser.'.format(parser))
                continue
            date_obj = parser_fnx()
            if self._is_valid_date_obj(date_obj):
                return date_obj
        else:
            return None
```
would do the trick. To test it, you can change the `PARSER` settings to something like this: `'PARSERS': ['aaaaa'] + default_parsers,`.

What do you think @Gallaecio ?

Feel free to adapt the warn message, as maybe we could mention the settings or whatever, and if you want, it could be a good idea to add a test with wrong parsers to see that all works as expected.

Reply:
Of course, if you prefer the "easier to ask for forgiveness than permission" version, you can go with:

```python
    def _parse(self):
        for parser in self._settings.PARSERS:
            try:
                date_obj = self._parsers[parser]()
            except KeyError:
                warn('"{}" is not a valid parser.'.format(parser))
                continue
            date_obj = parser_fnx()
            if self._is_valid_date_obj(date_obj):
                return date_obj
        else:
            return None
```

However, this would catch any `KeyError` occurring inside the parser function and we could generate false warns.

Reply:
I’m thinking that it may be better to raise an exception, instead of logging a warning. Warnings are good for deprecation, when something will stop working in the future. But in this case, I cannot think of a good reason why a user would pass the wrong parser, so I think we should raise an exception.

That said, it would be great to raise a `ValueError` with a message in line with the one you propose for the warning. I’ll update the changes accordingly and include a test as you suggested below.